dali/operators/reader/reader_op_test.cc

```diff
@@ -473,4 +475,279 @@ TEST(ReaderTestSimple, CheckNumSamples) {
   ASSERT_EQ(num_samples(3, 10), static_cast<Index>(std::ceil(10 * 1.0 / 3)));
 }
 
+class FileReaderTest : public DALITest {
+ public:
+  void SetUp() override {
+    constexpr int filecount = 16;
+
+    for (int i = 0; i < filecount; i++) {
+      std::string path = make_string("/tmp/dali_test_file_", to_string(i));
+      filepaths_.push_back(path);
+
+      // write a single byte to the file
+      std::ofstream f(path);
+      f << (uint8_t) i;
+    }
+  }
+
+  void TearDown() override {
+    for (const auto &path : filepaths_)
+      std::remove(path.data());
+  }
+
+  std::vector<std::string> filepaths_;
+
+  OpSpec MakeOpSpec() {
+      // Currently, only pad_last_batch=true is supported.
+      return OpSpec("FileReader")
+            .AddOutput("data_out", "cpu")
+            .AddOutput("labels", "cpu")
+            .AddArg("files", filepaths_)
+            .AddArg("checkpointing", true)
+            .AddArg("pad_last_batch", true);
+  }
+
+  void BuildPipeline(Pipeline &pipe) {
+    std::vector<std::pair<string, string>> outputs = {{"data_out", "cpu"}};
+    pipe.Build(outputs);
+  }
+
+  std::vector<uint8_t> RunEpoch(Pipeline &pipe, int batch_size,
+                                int num_shards = 1, bool stick_to_shard = false) {
+    std::vector<uint8_t> result;
+    int samples_per_shard = (filepaths_.size() + num_shards - 1) / num_shards;
+    int batches_per_shard = (samples_per_shard + batch_size - 1) / batch_size;
+    int iterations = (stick_to_shard ? 1 : num_shards) * batches_per_shard;
```

Comment:
Does the number of iterations in epoch depend on the stick_to_shard?

Reply:
Good catch! I misunderstood how the shards are working and indeed it was a bug (in both checkpointing and tests)

Reply:
I guess the `Reset()` workings are a bit misleading in that regard.